csharp/test/Apache.Arrow.Adbc.Tests/ClientTests.cs

```diff
@@ -268,7 +277,25 @@ static void AssertTypeAndValue(
                     }
                     else
                     {
-                        Assert.True(ctv.ExpectedValue.Equals(value), Utils.FormatMessage($"Expected value [{ctv.ExpectedValue}] does not match actual value [{value}] for {ctv.Name} for query [{query}]", environmentName));
+                        bool areEqual = false;
+
+                        if (value is ExpandoObject)
```

Comment:
Is this missing a case for when the result is an array?

Reply:
No. There is a check above `if (type.BaseType?.Name.Contains("PrimitiveArray") == false)` that puts us in a different path for comparing arrays.